src/main/scala/org/ergoplatform/nodeView/state/ErgoState.scala

```diff
@@ -40,35 +43,42 @@ trait ErgoState[IState <: MinimalState[ErgoPersistentModifier, IState]]
 
   /**
     * Tries to validate and execute transactions.
+    *
     * @return Result of transactions execution with total cost inside
     */
-  protected def execTransactionsTry(transactions: Seq[ErgoTransaction],
-                                    currentStateContext: ErgoStateContext)
-                                   (checkBoxExistence: ErgoBox.BoxId => Try[ErgoBox]): Try[Long] = {
+  protected def execTransactions(transactions: Seq[ErgoTransaction],
+                                 currentStateContext: ErgoStateContext)
+                                (checkBoxExistence: ErgoBox.BoxId => Try[ErgoBox]): ValidationResult[Long] = {
     import cats.implicits._
     implicit val verifier: ErgoInterpreter = ErgoInterpreter(currentStateContext.currentParameters)
-    def execTry(txs: List[ErgoTransaction], accCostTry: Try[Long]): Try[Long] = (txs, accCostTry) match {
-      case (tx :: tail, Success(accumulatedCost)) =>
-        val costTry = tx.statelessValidity.flatMap { _ =>
-          val boxesToSpendTry: Try[List[ErgoBox]] = tx.inputs.toList
-            .map(in => checkBoxExistence(in.boxId))
-            .sequence
-          val dataBoxesTry: Try[List[ErgoBox]] = tx.dataInputs.toList
-            .map(in => checkBoxExistence(in.boxId))
-            .sequence
-          boxesToSpendTry.flatMap { boxes =>
-            dataBoxesTry.flatMap { dataBoxes =>
-              tx.statefulValidity(boxes.toIndexedSeq, dataBoxes.toIndexedSeq, currentStateContext, accumulatedCost)
-            }
+    implicit val vs: ValidationSettings = ValidationRules.initialSettings
+
```

Comment:
The new execTx is not aborting on the first invalid transaction, which could be harmful (open to DoS attacks).

Reply:
It does not, the invalid transaction will lead to `r: Invalid`